runtime/caml/config.h

```diff
@@ -140,16 +140,19 @@ typedef unsigned char uint8_t;
 typedef long intnat;
 typedef unsigned long uintnat;
 #define ARCH_INTNAT_PRINTF_FORMAT "l"
+#define INTNAT_MIN LONG_MIN
```

Comment:
Have you tried moving these defines to `runtime/caml/misc.h`? `runtime/caml/config.h` doesn't include `<limits.h>` but these new macros depend on it, so it would make more sense to define them in a place where `<limits.h>` is included.

Reply:
Hmmm, it's true that `config.h` is missing `limits.h`, but adding the following to `misc.h` also seems like wasted duplication.
```c
#if SIZEOF_PTR == SIZEOF_LONG
/* Standard models: ILP32 or I32LP64 */
#define INTNAT_MIN LONG_MIN
#elif SIZEOF_PTR == SIZEOF_INT
/* Hypothetical IP32L64 model */
#define INTNAT_MIN INT_MIN
#elif SIZEOF_PTR == 8
/* Win64 model: IL32P64 */
#define INTNAT_MIN INT64_MIN
#endif
```
`config.h` could include `limits.h` instead, we've switched to C11, and most of the compatibility code around C99 integer types seems to have been added for old MSVC.

Reply:
The preprocessor logic duplication is unfortunate but probably acceptable (with a comment telling it must match what's in `config.h`) if adding `<limits.h>` to `config.h` is considered too large a change.

I think a `Changes` entry will be required if `config.h` now includes `<limits.h>`.

Reply:
I'm opting to add `limits.h` to `config.h`. I think a follow-up PR could switch entirely to C99 fixed-width integers all the macros and defines of `config.h`.